app/src/server/api/routers/clan.ts

```diff
@@ -1179,6 +1179,35 @@ export const clanRouter = createTRPCRouter({
       }
       return errorResponse(`Failed to initiate ${groupLabel} battle`);
     }),
+  instantJoinAndLead: protectedProcedure
+  .input(z.object({ clanId: z.string() }))
+  .output(baseServerResponse)
+  .mutation(async ({ ctx, input }) => {
+    const [user, fetchedClan] = await Promise.all([
+      fetchUser(ctx.drizzle, ctx.userId),
+      fetchClan(ctx.drizzle, input.clanId),
+    ]);
+    if (!fetchedClan) return errorResponse("Faction not found");
+    if (!user) return errorResponse("User not found");
+    if (!canEditClans(user.role)) return errorResponse("Permission denied");
+    if (user.clanId) return errorResponse("Already in a faction");
+
+    await ctx.drizzle.transaction(async (tx) => {
```

Comment:
We should avoid using transactions on the database; they work with planetscale, but I've consistently experienced when we introduce transactions we also start seeing deadlocks popping up. As for this specific case, I think simply putting the two transactions into a `Promise.all` is fine :)